lib/sqlalchemy/exc.py

```diff
@@ -56,10 +56,17 @@ def _message(self, as_unicode=compat.py3k):
         #
         if len(self.args) == 1:
             text = self.args[0]
+
+            if not isinstance(text, compat.binary_types + compat.string_types):
```

Comment:
hey there -

can we make this a little simpler, like this:

```
if not util.py2k or not as_unicode:
   text = str(self.args[0])
else:
   text = compat.text_type(text)
```

Reply:
Hello there,
I updated my changes - can you please check again?

Reply:
I just noticed this is already inside of an isinstance(), so let me just get the code onto gerrit where I can smooth the conditonals out some more

Reply:
Sure, play around with it :-)
Is there some contribution guidelines I can read, for the future?

Reply:
oh you did absolutely great, no worries.   contribution guide is at https://www.sqlalchemy.org/develop.html